packages/block-editor/src/hooks/child-layout.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	__experimentalUnitControl as UnitControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useSetting from '../components/use-setting';
+
+/**
+ * Inspector control panel containing the child layout related configuration
+ *
+ * @param {Object} props                        Block props.
+ * @param {Object} props.attributes             Block attributes.
+ * @param {Object} props.setAttributes          Function to set block attributes.
+ * @param {Object} props.__unstableParentLayout
+ *
+ * @return {WPElement} child layout edit element.
+ */
+export function ChildLayoutEdit( {
+	attributes,
+	setAttributes,
+	__unstableParentLayout: parentLayout,
+} ) {
+	const { style = {} } = attributes;
+	const { layout: childLayout = {} } = style;
+	const { selfStretch, flexSize } = childLayout;
+
+	return (
+		<>
+			<ToggleGroupControl
+				style={ { marginBottom: '8px' } }
+				label={ childLayoutOrientation( parentLayout ) }
+				value={ selfStretch || 'hug' }
+				onChange={ ( value ) => {
+					setAttributes( {
+						style: {
+							...style,
+							layout: {
+								...childLayout,
+								selfStretch: value,
+							},
+						},
+					} );
+				} }
+				isBlock={ true }
+			>
+				<ToggleGroupControlOption
+					key={ 'hug' }
+					value={ 'hug' }
+					label={ __( 'Hug' ) }
+				/>
+				<ToggleGroupControlOption
+					key={ 'fill' }
+					value={ 'fill' }
+					label={ __( 'Fill' ) }
+				/>
+				<ToggleGroupControlOption
+					key={ 'fixed' }
+					value={ 'fixed' }
+					label={ __( 'Fixed' ) }
+				/>
+			</ToggleGroupControl>
+			{ selfStretch === 'fixed' && (
+				<UnitControl
+					style={ { height: '36px' } }
```

Comment:
So this one's interesting. For some reason UnitControl has its height set to 100% and this doesn't play nicely with its grid parents here (the ToolsPanelItem parent of this component being a grid child, its height is automatically the full grid row height, so this was resulting in a 113px tall UnitControl).

Reply:
Very interesting! There's been some recent discussion about trying to avoid inline styles with `UnitControl` / `InputControl` components etc, (e.g here https://github.com/WordPress/gutenberg/pull/45340#issuecomment-1293805949). I wonder if it'd work to add a wrapper to the `UnitControl` (i.e. `Flex` or `HStack`?), or does that overly complicate things? Just copying in @mirka since there was that previous discussion about inline styles.

Reply:
Wrapping it in an HStack might fix the height issue, but isn't adding markup cruft worse than inline styles? 😬 Given it's a single component, we shouldn't need to wrap it at all.

Reply:
> but isn't adding markup cruft worse than inline styles?

Great question! I'd be curious to know what folks think about that. My understanding so far is that there are some components that expect its parent to be responsible for how it's laid out, so fill to the available space. But I'm not too sure which ones do and don't expect that

Reply:
I was just playing around with the `UnitControl` elsewhere, and it looks like the `__unstable-large` size option of `UnitControl` switches off the 100% height rule, and I see that's now being used in a lot of places. The rules associated with it [are found here](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/unit-control/styles/unit-control-styles.ts#L102). I was wondering if adding the following to the `UnitControl` might have the desired result?

`size={ '__unstable-large' }`

Not sure if that gets all the way there, though, since there's still a bunch of other styling rules included in the component 🤔

Reply:
After looking into this, I think I'd classify this as a bug in the way `Flex` is used in InputControl (#45877). I'll see if I can fix it up.

In the meantime, `auto` (rather than a hardcoded number) might be a better representation of our intent here:

```suggestion
					style={ { height: 'auto' } }
```

We can remove this hack once the root issue is fixed.

Reply:
I set it as 36px so the input would have the same height as the ToggleGroup above it, to match the designs from #44467. Otherwise there's quite a discrepancy:
<img width="268" alt="Screenshot 2022-11-21 at 11 32 04 am" src="https://user-images.githubusercontent.com/8096000/202935581-d714522c-c4a5-4c5c-8ab2-178683898204.png">
This is certainly not the best solution, but I'm not sure if there are plans to standardise the height of these controls? Cc @WordPress/gutenberg-design would it be ok to have unmatching heights until the issue is addressed globally?

Reply:
> I set it as 36px so the input would have the same height as the ToggleGroup above it, to match the designs from https://github.com/WordPress/gutenberg/issues/44467.

A quick comment on this until I get a chance to review this PR in full later today.

My understanding is that we should be trying to default controls to 40px height where possible as well as avoiding using inline styles on control components such as `UnitControl`.

To control the height we can use the component's `size` prop; e.g. `size="__unstable-large"`

Also, the linked design issue does have comments saying the height should be 40px not 36px (See https://github.com/WordPress/gutenberg/issues/44467#issuecomment-1261911573).

Reply:
That's correct. Worth sharing the grid we're working towards again here:

<img width="563" alt="Screenshot 2022-11-21 at 09 44 54" src="https://user-images.githubusercontent.com/846565/203018334-087d8a0b-e335-4472-a91f-d5f937ba1156.png">